src/containers/AddCardForm.js

```diff
@@ -0,0 +1,11 @@
+import { connect } from "react-redux";
+import { addCard } from "../actions";
+import AddCardForm from "../components/AddCard/AddCardForm/AddCardForm";
+
+
+const mapDispatchToProps = dispatch => ({
+    handleAddCard: (card) => dispatch(addCard(card.name, card.email)),
+})
+
+
+export default connect(null, mapDispatchToProps)(AddCardForm);
```

Comment:
Ahh, I guess you learned this pattern on the course.
It maybe looks nice in the theory, but in practice having component definition separate from `connect` makes problems for the following reasons:
- Let's say you are in the component and want to know which actual action creator is connected to the `this.props.handleAddCard`. You need to navigate to the other file `src/containers/AddCardForm` to find out, and navigate to the file with the action creator. So you have to go through 3 files instead of 2.
- You have to maintain more files and folder structures which are not giving you enough value for the effort.

Having separate component and `connect` makes sense in only 1 case. When the component is reusable and is connected to multiple different containers.
Let's say it's component "AddButton"
and it is connected in different places to `handleAdd: addCard`, `handleAdd: addPost`, `handleAdd: addUser`.
Otherwise, it's only added effort to have them separately.
This is my personal opinion, and some people do not agree with it. You also might not ;)

Reply:
good point, I didn't thought about this in this way